### PR TITLE
Add containerization for the ipa-tuura+Apache service

### PR DESCRIPTION
**Pull request by antoniotorresm**
Add Containerfile and podman-compose files that containerize ipa-tuura using httpd as HTTPS server. This replaces the previous Dockerfile.test, using Apache HTTPS server instead of the insecure built-in Django HTTP server.

By running the podman-compose.yaml, the container is built and deployed with the needed port mapping.

### CODE REVIEW

File: Containerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+FROM fedora:38
```
[the remark this answers is not in the document]
Comment (antoniotorresm) [reply Containerfile]: Thanks, updated.

**Comment by Tiboris** [comment]
Weird, the automation does not pick the PR

File: .github/workflows/gating.yaml [hunk 2]
```diff
@@ -64,7 +64,7 @@ jobs:
               ${{ ( github.event_name != 'pull_request' && github.ref == 'refs/heads/main' && format('{0}:latest', env.IMAGE_REGISTRY_QUAY) ) || '' }}
               ${{ ( github.event_name != 'pull_request' && github.ref_type == 'tag' && format('{0}:{1}', env.IMAGE_REGISTRY_QUAY, github.ref_name) ) || '' }}
           containerfiles: |
-            ./Dockerfile.test
+            ./Containerfile
```
Comment (f-trivino) [inline .github/workflows/gating.yaml:67]: I think we should keep using a Development mode container for gating workflow. Maybe we should improve the naming?

File: Dockerfile.test [hunk 3]
```diff
@@ -1,23 +0,0 @@
-FROM fedora:38
```
Comment (f-trivino) [inline Dockerfile.test:1]: I wouldn't delete this file, perhaps improve some aspects:

Comment (f-trivino) [inline Dockerfile.test:1]: FROM registry.fedoraproject.org/fedora:38

File: Dockerfile.test [hunk 4]
```diff
@@ -1,23 +0,0 @@
...
-python-devel openldap-devel python-ipalib \
-" && dnf install -y $INSTALL_PKGS && dnf clean all
-
-COPY . ipa-tuura
-
-WORKDIR /ipa-tuura
-
-RUN pip install -r src/install/requirements.txt
-
-RUN pip install dbus-python python-ldap
```
Comment (f-trivino) [inline Dockerfile.test:21]: these two should go to requirements.txt

File: Dockerfile.test [hunk 5]
```diff
@@ -1,23 +0,0 @@
...
-
-RUN python3 src/ipa-tuura/manage.py test
```
Comment (f-trivino) [inline Dockerfile.test:23]: this code line is important for gating, this is how we run the unit tests

File: podman-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+version: "3"
```
Comment (f-trivino) [inline podman-compose.yaml:1]: this file can be just an example on how to deploy in production mode, I think we should set ENVVARS in addition, check: 

https://github.com/freeipa/ipa-tuura/pull/43

File: src/conf/ipa.conf [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+[x509_extensions]
+# <SK> Used for (generically) custom field-value pairs that should be associated
+#   with the cert, such as extra DNS names, IP addresses, and emails
+subjectAltName = @alternate_names
+
+[alternate_names]
+# Extra domain names to associate with our cert
+#  - These can be a mix of wildcard, IP address, subdomain, etc.
+DNS.1 = *.ipa.test
```
Comment (f-trivino) [inline src/conf/ipa.conf:64]: we need to think how to handle these hard-coded values.

File: src/conf/ipa.conf [hunk 8]
```diff
@@ -0,0 +1,71 @@
+# Key:
```
Comment (f-trivino) [inline src/conf/ipa.conf:1]: perhaps we could delete all comments and keep just the settigns.

**Review by f-trivino — CHANGES_REQUESTED** [review]
thanks for the PR @antoniotorresm , I added few comments. I would also improve the commit message and include the issue ticket:

Related: https://github.com/freeipa/ipa-tuura/issues/29

I'm not sure if we are ready to switch everything to prod. This PR could be just the configuration files so that it is simple to deploy in prod until we figure out how to test this mode.

**Comment by Tiboris** [comment]
I think gating should use production Dockerfile whatever that means in `gating.yaml` definition we can later just exec the `python3 src/ipa-tuura/manage.py test` as a step after the container will be built. Or if we want to have both Dockerfiles or as you like Containerfiles we should share as many layers as possible from each other.

File: src/prod/Containerfile [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    unzip \
+    && dnf clean all
+RUN mkdir /www
+COPY . /www/ipa-tuura
+RUN pip install -r /www/ipa-tuura/src/install/requirements.txt
+
+# Setup ipa-tuura
+RUN echo 'LoadModule wsgi_module modules/mod_wsgi.so' >> /etc/httpd/conf/httpd.conf
+RUN sed -i 's/ALLOWED_HOSTS = \[\]/ALLOWED_HOSTS = \['"'*'"'\]/g' /www/ipa-tuura/src/ipa-tuura/root/settings.py
+RUN sed -i 's/staticfiles/static/g' /usr/local/lib/python3.11/site-packages/rest_framework_swagger/templates/rest_framework_swagger/index.html
```
Comment (f-trivino) [inline src/prod/Containerfile:45]: since swagger is not usable through HTTPs I think we can remove this line45.

File: src/prod/conf/ipa.conf [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+#   with the cert, such as extra DNS names, IP addresses, and emails
+subjectAltName = @alternate_names
+
+[alternate_names]
+# Extra domain names to associate with our cert
+#  - These can be a mix of wildcard, IP address, subdomain, etc.
+DNS.1 = *.ipa.test
+DNS.2 = localhost
+DNS.3 = 127.0.0.1
+DNS.4 = master.ipa.test
```
Comment (f-trivino) [inline src/prod/conf/ipa.conf:67]: I know that we will improve this when enabling extra automation such as certbot, but let's change "master.ipa.test" by "bridge.ipa.test". Service is supposed to be deployed in a separate domain as ipa domain.

File: src/prod/podman-compose.yaml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+version: "3"
+services:
+  tuura-server:
+    container_name: tuura-server
+    build:
+      context: ../..
+      dockerfile: src/prod/Containerfile
+    environment:
+      - DJANGO_SECRET_KEY="^$z^y$^ndlem@_f1)($_5vye6t!dk#8+8&9=y5*=-r(v465xg+"
+      - DJANGO_DEBUG='True'
```
Comment (f-trivino) [inline src/prod/podman-compose.yaml:10]: DJANGO_DEBUG='False' is intended for production.

**Review by f-trivino — COMMENTED** [review]
@antoniotorresm added few comments.

**Comment by antoniotorresm** [comment]
Thanks, pushed requested changes.

**Review by f-trivino — APPROVED** [review]
LGTM.

**Pull request merged**